app/views/editor/rich.html.erb

```diff
@@ -414,11 +423,43 @@
       }
     });
 
+    <% if @lat and @lon %>
+      <% if @zoom %>
+        editor.mapModule.blurredLocation.setZoom(<%= @zoom %>);
+      <% else %>
+        let zoom = editor.mapModule.blurredLocation.getZoomFromCoordinates(<%= @lat %>, <%= @lon %>);
+        addTag('zoom:'+zoom, '/profile/tags/create/<%= current_user.uid %>');
```

Comment:
Why we are explicitly adding tag here? 
Pl.editor will automatically add zoom tag, right?
Moreover if user changes zoom level, we will have 2 zoom tags? One from here and other from PL.editor.

Reply:
This only calculates zoom in the instances that the user has a lat and lon in their profile but no zoom level saved. So it calculates a zoom and saves it to their profile using the addTag javascript function (I worked on that a few weeks ago). It does not add the tag into the map, just to their profile. :)

Reply:
Oh i see, the name is misleading. Very cool place to put this though :)